scipy/linalg/interpolative.py

```diff
@@ -424,65 +433,45 @@ def _is_real(A):
 
 def seed(seed=None):
     """
-    Seed the internal random number generator used in this ID package.
-
-    The generator is a lagged Fibonacci method with 55-element internal state.
-
-    Parameters
-    ----------
-    seed : int, sequence, 'default', optional
-        If 'default', the random seed is reset to a default value.
-
-        If `seed` is a sequence containing 55 floating-point numbers
-        in range [0,1], these are used to set the internal state of
-        the generator.
-
-        If the value is an integer, the internal state is obtained
-        from `numpy.random.RandomState` (MT19937) with the integer
-        used as the initial seed.
-
-        If `seed` is omitted (None), ``numpy.random.rand`` is used to
-        initialize the generator.
+    This function used to set the seed of the randomization algorithms used in the
+    `scipy.linalg.interpolative` functions written in Fortran77.
 
+    The library has been ported to Python and now the functions use the native NumPy
+    generators and this function has no content and returns None. Thus this function
+    should not be used and will be removed in SciPy version 1.17.0.
     """
-    # For details, see :func:`_backend.id_srand`, :func:`_backend.id_srandi`,
-    # and :func:`_backend.id_srando`.
-
-    if isinstance(seed, str) and seed == 'default':
-        _backend.id_srando()
-    elif hasattr(seed, '__len__'):
-        state = np.asfortranarray(seed, dtype=float)
-        if state.shape != (55,):
-            raise ValueError("invalid input size")
-        elif state.min() < 0 or state.max() > 1:
-            raise ValueError("values not in range [0,1]")
-        _backend.id_srandi(state)
-    elif seed is None:
-        _backend.id_srandi(np.random.rand(55))
-    else:
-        rnd = np.random.RandomState(seed)
-        _backend.id_srandi(rnd.rand(55))
+    warnings.warn("`scipy.linalg.interpolative.seed` is deprecated and will be "
+                  "removed in SciPy 1.17.0.", DeprecationWarning, stacklevel=2)
```

Comment:
This is a no-op function now. The stacklevel should be corrected and should be added to the deprecation tracker.